typescript/cli/src/check/warp.ts

```diff
@@ -1,19 +1,66 @@
 import { stringify as yamlStringify } from 'yaml';
 
-import { WarpRouteDeployConfig, normalizeConfig } from '@hyperlane-xyz/sdk';
-import { ObjectDiff, diffObjMerge } from '@hyperlane-xyz/utils';
+import {
+  HypTokenRouterConfig,
+  WarpRouteDeployConfig,
+  sortArraysInConfig,
+} from '@hyperlane-xyz/sdk';
+import {
+  FormatObjectFormatter,
+  ObjectDiff,
+  diffObjMerge,
+  formatObj,
+} from '@hyperlane-xyz/utils';
 
 import { log, logGreen } from '../logger.js';
 import { formatYamlViolationsOutput } from '../utils/output.js';
 
+const formatter: FormatObjectFormatter = (
+  obj: any,
+  propPath: ReadonlyArray<string>,
+) => {
+  // Needed to check if we are currently inside the remoteRouters object
+  const maybeRemoteRoutersKey = propPath[propPath.length - 3];
```

Comment:
Don't we need to bound checking? I.e. what if the path is only 2 elements, then it will be -1, thus it will check for the last element?

Reply:
If an element does not exist at a given index, we will get undefined as value, it will behave like this even with negative indexes